src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -0,0 +1,773 @@
+        {
+            var paramNames = new List<string>();
+            foreach (var passThruParameter in methodParameters)
+            {
+                if (testMethodParameters.Contains(passThruParameter))
+                {
+                    paramNames.Add(passThruParameter.Name);
+                    continue;
+                }
+                string? paramName = null;
+                foreach (ExampleParameter exampleParameter in exampleModel.MethodParameters)
+                {
+                    if (passThruParameter.Name == exampleParameter.Parameter.CSharpName())
+                    {
+                        paramName = WriteExampleParameterDeclaration(_writer, exampleParameter, passThruParameter);
+                    }
+                }
+                if (paramName is null)
+                {
+                    paramName = useVariableName(passThruParameter.Name);
+                    if (passThruParameter.ValidateNotNull)
+                    {
+                        _writer.Line($"{passThruParameter.Type} {paramName} = null; /* Can't find this parameter in example, please provide value here!*/");
+                    }
+                    else
+                    {
+                        _writer.Line($"{passThruParameter.Type} {paramName} = null;");
+                    }
+                }
+                paramNames.Add(paramName);
+            }
+            return paramNames;
+        }
+
+        public CSharpType GenResponseType(MgmtClientOperation clientOperation, bool async, string? methodName = null)
+        {
+            if (clientOperation.IsLongRunningOperation() || methodName == "CreateOrUpdate")
+            {
+                var lroObjectType = GetLROObjectType(clientOperation.First().Operation, async);
+                return lroObjectType.WrapAsync(async);
+            }
+            else if (clientOperation.IsPagingOperation(Context))
+            {
+                var itemType = clientOperation.First(restOperation => restOperation.IsPagingOperation(Context)).GetPagingMethod(Context)!.PagingResponse.ItemType;
+                var actualItemType = WrapResourceDataType(itemType, clientOperation.First())!;
+                return actualItemType.WrapPageable(async);
+            }
+            else if (clientOperation.IsListOperation(Context, out var itemType) && methodName != "Get")
+            {
+                var returnType = new CSharpType(typeof(IReadOnlyList<>), WrapResourceDataType(itemType, clientOperation.First())!);
+                return GetResponseType(returnType, async);
+            }
+            else
+            {
+                var returnType = WrapResourceDataType(clientOperation.ReturnType, clientOperation.First());
+                return GetResponseType(returnType, async);
+            }
+        }
+
+        public string? ParseRequestPath(MgmtTypeProvider tp, string requestPath, ExampleModel exampleModel)
```

Comment:
We provide similar APIs in the `RequestPath` class, therefore we should no longer need to parse the request path by its raw string.